src/linux/network.rs

```diff
@@ -60,6 +60,73 @@ impl Networks {
     }
 }
 
+fn refresh_networks_list_from_sysfs(
+    interfaces: &mut HashMap<String, NetworkData>,
+    sysfs_net: &Path,
+) {
+    if let Ok(dir) = std::fs::read_dir(sysfs_net) {
+        let mut data = vec![0; 30];
+
+        let entries: Vec<DirEntry> = dir
+            .flatten()
+            .filter(|e| e.file_name().into_string().is_ok())
+            .collect();
+        let names: Vec<String> = entries
+            .iter()
+            .filter_map(|e| e.file_name().into_string().ok())
+            .collect();
+
+        // Remove interfaces that are gone
+        for name in interfaces
+            .keys()
+            .filter(|n| !names.contains(n))
+            .map(|n| n.to_owned())
+            .collect::<Vec<String>>()
```

Comment:
```suggestion
        for name in interfaces
            .keys()
            .filter(|n| !names.contains(n))
            .map(|n| n.to_owned())
```

(no need for the `collect` here)

Reply:
I this this is needed because otherwise we would be iterating and modifying the map at the time.
Without it I get this error:
```
error[E0502]: cannot borrow `*interfaces` as mutable because it is also borrowed as immutable
  --> src/linux/network.rs:85:13
   |
80 |           for name in interfaces
   |                       ----------
   |                       |
   |  _____________________immutable borrow occurs here
   | |
81 | |             .keys()
82 | |             .filter(|n| !names.contains(n))
83 | |             .map(|n| n.to_owned())
   | |__________________________________- immutable borrow later used here
84 |           {
85 |               interfaces.remove(&name);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
```

Reply:
Oh indeed. Well, the new approach I suggested would allow us to get around this issue. :)